OpenSim/Simulation/SimbodyEngine/Joint.cpp

```diff
@@ -640,6 +629,26 @@ void Joint::updateFromXMLNode(SimTK::Xml::Element& aNode, int versionNumber)
                 childNameElt->setValue(childFrameName + "_offset");
             }
         }
+
+        // Version 30506 replaced Joint's CoordinateSet with a "coordinates"
+        // list property.
+        if (documentVersion <= 30506) {
+            if (aNode.hasElement("CoordinateSet")) {
+                auto coordSetIter = aNode.element_begin("CoordinateSet");
+                auto coordIter = coordSetIter->getRequiredElement("objects")
```

Comment:
> If w ehave a WeldJoint, will it have an empty "CoordinateSet"? Maybe check that coordSetIter is not at end before dereferencing?

AFAIK WeldJoints can never have a "CoordinateSet" element because they have no Coordinates and, therefore, cannot have any non-default Coordinate-related properties to serialize. Line 636 ensures the "CoordinateSet" element exists; if it does, I believe the XML schema (if we had one) would demand that it holds a non-empty "objects" element (hence my use of `getRequiredElement()`).